src/main/java/org/craftercms/engine/graphql/impl/fetchers/ContentTypeBasedDataFetcher.java

```diff
@@ -261,64 +274,124 @@ protected void processSelection(String path, Selection currentSelection, BoolQue
             case ARG_NAME_MATCHES:
-                return QueryBuilders.matchQuery(fieldPath, getRealValue(filter.getValue(), env));
+                return matchQuery(fieldPath, getRealValue(filter.getValue(), env));
             case ARG_NAME_REGEX:
-                return QueryBuilders.regexpQuery(fieldPath, getRealValue(filter.getValue(), env).toString());
+                return regexpQuery(fieldPath, getRealValue(filter.getValue(), env).toString());
             case ARG_NAME_LT:
-                return QueryBuilders.rangeQuery(fieldPath).lt(getRealValue(filter.getValue(), env));
+                return rangeQuery(fieldPath).lt(getRealValue(filter.getValue(), env));
             case ARG_NAME_GT:
-                return QueryBuilders.rangeQuery(fieldPath).gt(getRealValue(filter.getValue(), env));
+                return rangeQuery(fieldPath).gt(getRealValue(filter.getValue(), env));
             case ARG_NAME_LTE:
-                return QueryBuilders.rangeQuery(fieldPath).lte(getRealValue(filter.getValue(), env));
+                return rangeQuery(fieldPath).lte(getRealValue(filter.getValue(), env));
             case ARG_NAME_GTE:
-                return QueryBuilders.rangeQuery(fieldPath).gte(getRealValue(filter.getValue(), env));
+                return rangeQuery(fieldPath).gte(getRealValue(filter.getValue(), env));
             case ARG_NAME_EXISTS:
                 boolean exists = (boolean) getRealValue(filter.getValue(), env);
                 if (exists) {
-                    return QueryBuilders.existsQuery(fieldPath);
+                    return existsQuery(fieldPath);
+                } else {
+                    return boolQuery().mustNot(existsQuery(fieldPath));
+                }
+            default:
+                // never happens
+                return null;
+        }
+    }
+
+    protected QueryBuilder getFilterQueryFromMapEntry(String fieldPath, Map.Entry<String, Object> filter) {
+        switch (filter.getKey()) {
+            case ARG_NAME_EQUALS:
+                return termQuery(fieldPath, filter.getValue());
+            case ARG_NAME_MATCHES:
+                return matchQuery(fieldPath, filter.getValue());
+            case ARG_NAME_REGEX:
+                return regexpQuery(fieldPath, filter.getValue().toString());
+            case ARG_NAME_LT:
+                return rangeQuery(fieldPath).lt(filter.getValue());
+            case ARG_NAME_GT:
+                return rangeQuery(fieldPath).gt(filter.getValue());
+            case ARG_NAME_LTE:
+                return rangeQuery(fieldPath).lte(filter.getValue());
+            case ARG_NAME_GTE:
+                return rangeQuery(fieldPath).gte(filter.getValue());
+            case ARG_NAME_EXISTS:
+                boolean exists = (boolean) filter.getValue();
+                if (exists) {
+                    return existsQuery(fieldPath);
                 } else {
-                    return QueryBuilders.existsQuery(fieldPath);
```

Comment:
Asked about this change.
This is a fix to a different bug, discovered during testing. It's not related to the feature.